packages/ai/src/to-model-output.ts

```diff
@@ -0,0 +1,373 @@
+import type { ToolCallResult, ToolCallSuccess, ToolCallFailure } from '@tool2agent/types';
+
+/**
+ * Text output for the language model.
+ */
+export type ModelOutputText = { type: 'text'; value: string };
+
+/**
+ * JSON output for the language model.
+ */
+export type ModelOutputJson = { type: 'json'; value: unknown };
+
+/**
+ * Error text output for the language model.
+ */
+export type ModelOutputErrorText = { type: 'error-text'; value: string };
+
+/**
+ * Error JSON output for the language model.
+ */
+export type ModelOutputErrorJson = { type: 'error-json'; value: unknown };
+
+/**
+ * Text content in a multipart model output.
+ */
+export type ContentPartText = { type: 'text'; text: string };
+
+/**
+ * Media content in a multipart model output.
+ */
+export type ContentPartMedia = {
+  type: 'media';
+  /** Base-64 encoded media data */
+  data: string;
+  /** IANA media type (e.g., 'image/png', 'image/jpeg') */
+  mediaType: string;
+};
+
+/**
+ * Multipart content output for the language model.
+ */
+export type ModelOutputContent = {
+  type: 'content';
+  value: Array<ContentPartText | ContentPartMedia>;
+};
+
+/**
+ * The output type that toModelOutput can return.
+ * This is compatible with the AI SDK's LanguageModelV2ToolResultPart['output'].
+ */
+export type ModelOutput =
+  | ModelOutputText
+  | ModelOutputJson
+  | ModelOutputErrorText
+  | ModelOutputErrorJson
+  | ModelOutputContent;
+
+/**
+ * A function that renders an OutputType to a model output.
+ * Can return various formats depending on the use case.
+ */
+export type OutputRenderer<OutputType> = (output: OutputType) => ModelOutput;
+
+/**
+ * A function that renders a failure result to a model output.
+ * If not provided, a default text representation will be used.
+ */
+export type FailureRenderer<InputType> = (failure: ToolCallFailure<InputType>) => ModelOutput;
+
+/**
+ * Options for creating a toModelOutput function.
+ */
+export type CreateToModelOutputOptions<InputType, OutputType> = {
+  /**
+   * Renderer for successful results.
+   * Takes the OutputType value from the success result.
+   * If not provided, falls back to JSON representation.
+   */
+  renderOutput?: OutputRenderer<OutputType>;
+  /**
+   * Renderer for failure results.
+   * If not provided, uses a default text representation of the failure.
+   */
+  renderFailure?: FailureRenderer<InputType>;
+  /**
+   * Whether to include feedback and instructions in the output.
+   * Default is true.
+   */
+  includeFeedback?: boolean;
+  /**
+   * Whether to include instructions in the output.
+   * Default is true.
+   */
+  includeInstructions?: boolean;
+};
+
```

Comment:
this is too complicated. let the user provide a function that just converts the result to model output